pkg/config/analysis/analyzer.go

```diff
@@ -51,17 +52,41 @@ func (c *CombinedAnalyzer) Metadata() Metadata {
 
 // Analyze implements Analyzer
 func (c *CombinedAnalyzer) Analyze(ctx Context) {
+	// if SchemaValidation is requested, run it first, removing all flagged
+	// resources from future analysis, to prevent panics
 	for _, a := range c.analyzers {
-		scope.Analysis.Debugf("Started analyzer %q...", a.Metadata().Name)
-		if ctx.Canceled() {
-			scope.Analysis.Debugf("Analyzer %q has been cancelled...", c.Metadata().Name)
+		if _, ok := a.(*schema.ValidationAnalyzer); ok {
+			if runAnalyzer(ctx, a, c.Metadata().Name) {
+				return
+			}
+		}
+	}
+	for _, msg := range ctx.(MutableContext).Messages() {
+		ctx.(MutableContext).Skip(msg.Resource)
```

Comment:
why do we skip all of these?

Reply:
Because the resources are invalid, they may have nil values for non-nullable fields, or other malformations which can cause analyzers to panic.  The alternative is to expand analyzers' logic to also handle invalid resources...

Reply:
It looks like it just iterates over `ctx.Messages()` which sounds like *all* messages though...?

Also instead of skip, why not just remove them?

Reply:
This is all messages so far, and since all we've run so far is the validation analyzers, it will be 100% validation messages.

Why skip instead of remove?  The analyzer context can be backed either by a kubernetes API or by files containing istio resources.  In either case there is no meaningful way to safely remove resources.  We certainly wouldn't want to delete these resources from either kubernetes or the file system.  Skipping them ensures that no non-validation analyzers will evaluate these invalid resources.